analyticsdataserver/router.py

```diff
@@ -2,28 +2,112 @@
+            #    for those models created in the previous migration. The write process triggers the db_for_write method
+            #    of Django routers. The db_for_write method returns the value of
+            #    getattr(settings, 'ANALYTICS_DATABASE', 'default'), which writes the data to the wrong database. There
+            #    is no straight forward way to force the db_for_write method to return the value of
+            #    getattr(settings, 'ANALYTICS_DATABASE_V1') instead. We forgo migrating that app entirely
+            #    in this database.
+            if app_label == 'enterprise_data' and database == getattr(settings, 'ANALYTICS_DATABASE_V1'):
+                return False
+
+            # If the requested database for migrations is the default database, but other analytics
+            # databases are set up, they should be favored over the default database, so prevent these
+            # migrations from being run on the default database.
+            # We only allow migrations on analytics application models on the default database when analytics
+            # is not set up.
+            if database == 'default' and 'default' in databases and len(databases) > 1:
+                return False
+
+            return True
+
+        # Ideally, we'd only want to migrate other applications into the default database to better
+        # mimic production. However, the enterprise_data application has the migration
+        # 0018_enterprisedatafeaturerole_enterprisedataroleassignment, which creates a model with a ForeignKey field
+        # pointing to the auth_user model. MySQL does not allow cross-database relations, because they violate
+        # referential integrity, so we cannot only migrate auth_user into the default database.
+        # For that reason, we need to allow the migration of the auth_user table into the ANALYTICS_DATABASE
+        # as well.
+        # Please see https://docs.djangoproject.com/en/3.2/topics/db/multi-db/#limitations-of-multiple-databases.
+        return database in ['default', getattr(settings, 'ANALYTICS_DATABASE')]
+
+
+class AnalyticsAPIRouter:
+    """
+    This router's role is to route database operations to the appropriate database when there is an 'analytics_database'
+    "hint" in the local thread data. This "hint" is set by the RequestVersionMiddleware and is meant to route
+    database operations to particular databases depending on the version of the API requested via the view.
+    """
     def _get_database(self, app_label):
-        if app_label in ('v0', 'v1', 'enterprise_data'):
+        if app_label in ANALYTICS_APP_LABELS:
             if hasattr(thread_data, 'analyticsapi_database'):
                 return getattr(thread_data, 'analyticsapi_database')
+            # If there is no analyticsapi_database database hint, then fall back to next router.
+            return None
+        return None
```

Comment:
could collapse this a little to be more obviously as simple as it is

if analytics app and hasattr
return getattr
return none